src/systems/buoyancy_engine/BuoyancyEngine.cc

```diff
@@ -77,8 +80,32 @@ class ignition::gazebo::systems::BuoyancyEnginePrivateData
 
   /// \brief mutex for protecting bladder volume and set point.
   public: std::mutex mtx;
+
+  /// \brief get fluid density
+  public: double CurrentFluidDensity(
+    EntityComponentManager &_ecm,
+    const Link &_link) const;
 };
 
+//////////////////////////////////////////////////
+double BuoyancyEnginePrivateData::CurrentFluidDensity(
+  EntityComponentManager &_ecm,
+  const Link &_link) const
+{
+  if (!this->surface.has_value())
+    return fluidDensity;
+
+  auto pose = _link.WorldPose(_ecm);
+  if (!pose.has_value())
+    return fluidDensity;
+
+  if(pose->Pos().Z() < this->surface.value())
```

Comment:
```suggestion
  if (pose->Pos().Z() < this->surface.value())
```

I wonder why the linter is not catching these